Bug: Error executing this data block https://registry.terraform.io/providers/mulesoft-anypoint/anypoint/latest/docs/data-sources/amq

module.amq.data.anypoint_amq.dlq: Reading...

Planning failed. Terraform encountered an error while generating this plan.

╷
│ Error: Request cancelled
│ 
│   with module.amq.data.anypoint_amq.dlq,
│   on ../../main.tf line 1, in data "anypoint_amq" "dlq":
│    1: data "anypoint_amq" "dlq" {
│ 
│ The plugin.(*GRPCProvider).ReadDataSource request was cancelled.
╵

Stack trace from the terraform-provider-anypoint_v1.8.2 plugin:

panic: invalid element type: schema.ValueType

goroutine 81 [running]:
[github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.serializeCollectionMemberForHash](http://github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.serializeCollectionMemberForHash)(0x0?, {0x1032e07c0?, 0x1400090e510?}, {0x103354300?, 0x103230d48?})
        [github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/serialize.go:128](http://github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/serialize.go:128) +0xfc
[github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.SerializeValueForHash](http://github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.SerializeValueForHash)(0x1400088e390, {0x1032c8ee0, 0x140000b6a98}, 0x14000323b80)
        [github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/serialize.go:33](http://github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/serialize.go:33) +0x5e4
[github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.SerializeResourceForHash](http://github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.SerializeResourceForHash)(0x1400088e390, {0x103373420?, 0x14000902ea0}, 0x140002c5d20?)
        [github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/serialize.go:115](http://github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/serialize.go:115) +0x270
[github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema](http://github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema).(*Schema).ZeroValue.HashResource.func2({0x103373420?, 0x14000902ea0?})
        [github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/set.go:33](http://github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/set.go:33) +0x40
[github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema](http://github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema).(*Set).hash(0x14000323b80?, {0x103373420?, 0x14000902ea0?})
        [github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/set.go:218](http://github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/set.go:218) +0x34
[github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema](http://github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema).(*Set).add(0x140002c5d20, {0x103373420, 0x14000902ea0}, 0x0)
        [github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/set.go:198](http://github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/set.go:198) +0x64
[github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema](http://github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema).(*ConfigFieldReader).readSet(0x14000902d20, {0x1400090e1d0, 0x1, 0x1}, 0x14000323cc0)
        [github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/field_reader_config.go:284](http://github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/field_reader_config.go:284) +0x270
[github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema](http://github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema).(*ConfigFieldReader).readField(0x14000902d20, {0x1400090e1d0, 0x1, 0x1}, 0x0)
        [github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/field_reader_config.go:107](http://github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/field_reader_config.go:107) +0x2e8
[github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema](http://github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema).(*ConfigFieldReader).ReadField(0x103e753e8?, {0x1400090e1d0?, 0x1030b5bfd?, 0x140002c5cc0?})
        [github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/field_reader_config.go:29](http://github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/field_reader_config.go:29) +0x7c
[github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema](http://github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema).(*MultiLevelFieldReader).ReadFieldExact(0x1400090e1c0?, {0x1400090e1d0, 0x1, 0x1}, {0x1030b5bfd, 0x6})
        [github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/field_reader_multi.go:31](http://github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/field_reader_multi.go:31) +0x70
[github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema](http://github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema).(*ResourceData).get(0x14000625d00, {0x1400090e1d0, 0x1, 0x1}, 0x0?)
        [github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/resource_data.go:548](http://github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/resource_data.go:548) +0xf0
[github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema](http://github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema).(*ResourceData).getChange(0x14000625d00, {0x1030b5c57?, 0x6?}, 0x1?, 0x12)
        [github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/resource_data.go:524](http://github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/resource_data.go:524) +0x148
[github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema](http://github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema).(*ResourceData).diffChange(0x0?, {0x1030b5c57?, 0x9?})
        [github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/resource_data.go:501](http://github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/resource_data.go:501) +0x44
[github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.diffSet](http://github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.diffSet)(0x140002eea80, {0x1034cbca8, 0x140009020c0}, {0x1030b5c57, 0x6}, 0x14000323cc0, 0x1400088f0b8, {0x1034cf330, 0x14000625d00}, 0x0)
        [github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/schema.go:1413](http://github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/schema.go:1413) +0x5c
[github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.diff](http://github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.diff)(0x140002eea80, {0x1034cbca8, 0x140009020c0}, {0x1030b5c57, 0x6}, 0x14000323cc0, 0x14000625c80, {0x1034cf330, 0x14000625d00}, 0x0)
        [github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/schema.go:1135](http://github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/schema.go:1135) +0x180
[github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.Diff](http://github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.Diff)(0x140002eea80, {0x1034cbca8, 0x140009020c0}, 0x0, 0x14000902ae0, 0x0, {0x1034a91a0, 0x140000ce000}, 0x1)
        [github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/schema.go:679](http://github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/schema.go:679) +0x258
[github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema](http://github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema).(*Resource).Diff(0x14000326000, {0x1034cbca8, 0x140009020c0}, 0x0, 0x14000902ae0, {0x1034a91a0, 0x140000ce000})
        [github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/resource.go:868](http://github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/resource.go:868) +0xcc
[github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema](http://github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema).(*GRPCProviderServer).ReadDataSource(0x14000017b30, {0x1034cbca8?, 0x1400077ff50?}, 0x140002c4a00)
        [github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/grpc_provider.go:1166](http://github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/grpc_provider.go:1166) +0x228
[github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server](http://github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server).(*server).ReadDataSource(0x140003b4320, {0x1034cbca8?, 0x1400077f770?}, 0x14000698460)
        [github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/tf5server/server.go:658](http://github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/tf5server/server.go:658) +0x2cc
[github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadDataSource_Handler](http://github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadDataSource_Handler)({0x103467da0, 0x140003b4320}, {0x1034cbca8, 0x1400077f770}, 0x1400061eee0, 0x0)
        [github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:421](http://github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:421) +0x1b8
[google.golang.org/grpc](http://google.golang.org/grpc).(*Server).processUnaryRPC(0x140001883c0, {0x1034d0ae0, 0x1400039d380}, 0x140000ce480, 0x1400028c630, 0x103d23e90, 0x0)
        [google.golang.org/grpc@v1.48.0/server.go:1295](http://google.golang.org/grpc@v1.48.0/server.go:1295) +0x870
[google.golang.org/grpc](http://google.golang.org/grpc).(*Server).handleStream(0x140001883c0, {0x1034d0ae0, 0x1400039d380}, 0x140000ce480, 0x0)
        [google.golang.org/grpc@v1.48.0/server.go:1636](http://google.golang.org/grpc@v1.48.0/server.go:1636) +0x720
[google.golang.org/grpc](http://google.golang.org/grpc).(*Server).serveStreams.func1.2()
        [google.golang.org/grpc@v1.48.0/server.go:932](http://google.golang.org/grpc@v1.48.0/server.go:932) +0x78
created by [google.golang.org/grpc](http://google.golang.org/grpc).(*Server).serveStreams.func1 in goroutine 43
        [google.golang.org/grpc@v1.48.0/server.go:930](http://google.golang.org/grpc@v1.48.0/server.go:930) +0x1e4

Error: The terraform-provider-anypoint_v1.8.2 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

--- a/anypoint/data_source_ame.go
+++ b/anypoint/data_source_ame.go
@@ -73,7 +73,9 @@ func dataSourceAME() *schema.Resource {
 							Type:        schema.TypeList,
 							Optional:    true,
 							Description: "Includes only results with the given Ids.",
-							Elem:        schema.TypeString,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
 						},
 					},
 				},

--- a/anypoint/data_source_amq.go
+++ b/anypoint/data_source_amq.go
@@ -72,7 +72,9 @@ func dataSourceAMQ() *schema.Resource {
 							Type:        schema.TypeList,
 							Optional:    true,
 							Description: "Includes only results with the given Ids.",
-							Elem:        schema.TypeString,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
 						},
 					},
 				},

--- a/anypoint/data_source_fabrics.go
+++ b/anypoint/data_source_fabrics.go
@@ -127,7 +127,9 @@ var FabricsIngressDomainsDefinition = &schema.Resource{
 			Type:        schema.TypeList,
 			Computed:    true,
 			Description: "The list of domains.",
-			Elem:        schema.TypeString,
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+			},
 		},
 	},
 }

